### PR TITLE
Change folder breadcrumb on folder change in general settings

### LINKED ISSUE #85547: When changing the dashboard folder, the breadcrumb dashboard disappear
When you change a folder, the breadcrumbs do not reflect that, the previous folder is removed and the breadcrumb appears as if the dashboard was removed from a folder. See the attached video:

https://github.com/grafana/grafana/assets/2376619/75c96c09-2483-486b-a1e4-796baef8f839

### PR DESCRIPTION
**Pull request by mdvictor**
Fixes a bug where the folder breadcrumb no longer appears after changing the folder from a dashboards general settings. This happens because not all folders in the general settings select box are actually stored in the `navIndex`, so when this changes the dashboard scene `getPageNav` method doesn't find the new folder and returns `not-found`. This solution updates the `navIndex` to include selected folder in it so that the dashboard can pick it up easily and render it in the breadcrumbs.

Permissions are accounted for by the folders select box in the settings page so we don't need to worry about that.

https://github.com/grafana/grafana/assets/36818606/5a3ab752-c0dd-40ad-8b3b-0b87912c090b

Fixes https://github.com/grafana/grafana/issues/85547

### CODE REVIEW

File: public/app/features/dashboard-scene/settings/GeneralSettingsEditView.tsx [hunk 1]
```diff
@@ -96,13 +97,18 @@ export class GeneralSettingsEditView
     this._dashboard.setState({ tags: value });
   };
 
-  public onFolderChange = (newUID: string | undefined, newTitle: string | undefined) => {
+  public onFolderChange = async (newUID: string | undefined, newTitle: string | undefined) => {
```
Comment (ivanortegaalba) [inline public/app/features/dashboard-scene/settings/GeneralSettingsEditView.tsx:100]: Great finding! I'd suggest adding some tests here to make sure we call the updateNavModel with the new folder ID. WDYT?

File: public/app/features/dashboard-scene/pages/DashboardScenePageStateManager.ts [hunk 2]
```diff
@@ -228,6 +228,15 @@ export class DashboardScenePageStateManager extends StateManagerBase<DashboardSc
     return null;
   }
 
+  public async updateNavModel(folderUid: string) {
```
Comment (ivanortegaalba) [inline public/app/features/dashboard-scene/pages/DashboardScenePageStateManager.ts:231]: What do you think about updating signature for [`initNavModel`](https://github.com/grafana/grafana/pull/86342/files#diff-014099493438b3f54a14bf4d12e496c65d45fc3dd9b8a91f27a0d1571f538015L231) to keep consistent the function signatures?

Also, I'd add some tests here. WDYT?

**Review by ivanortegaalba — APPROVED** [review]
It works like a charm! Good work Victor!

File: public/app/features/dashboard-scene/pages/DashboardScenePageStateManager.ts (hunk 2 above)
Comment (mdvictor) [reply public/app/features/dashboard-scene/pages/DashboardScenePageStateManager.ts:231]: In that case we might as well cut one of the functions and use the same everywhere, which is what I'm going for now :D

File: public/app/features/dashboard-scene/settings/GeneralSettingsEditView.tsx (hunk 1 above)
Comment (mdvictor) [reply public/app/features/dashboard-scene/settings/GeneralSettingsEditView.tsx:100]: adapted tests for this

**Comment by mdvictor** [comment]
@ivanortegaalba I refactored things a bit. Calling the `DashboardScenePageStateManager` in the settings somehow caused all tests in `SaveDashboardDrawer.test.tsx` to fail because of some redux reducers being called badly at some point. Couldn't figure out why or how things connected tho so I moved the `updateNavModel` function in a `utils` file. If you have a better idea on keeping this in the state manager let me know!

**Pull request merged**